components/LedgerRegistrationModal.tsx

```diff
@@ -0,0 +1,268 @@
+import { Ionicons } from '@expo/vector-icons';
+import React, { useState } from 'react';
+import {
+  Modal,
+  Pressable,
+  ScrollView,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+
+interface LedgerRegistrationModalProps {
+  visible: boolean;
+  onClose: () => void;
+  onConfirm: (data: LedgerRegistrationData) => void;
+}
+
+interface LedgerRegistrationData {
+  name: string;
+  phoneNumber: string;
+  initialAmount: string;
+  receptionist: string;
+  paymentMethod: string;
+}
+
+export default function LedgerRegistrationModal({
+  visible,
+  onClose,
+  onConfirm,
+}: LedgerRegistrationModalProps) {
+  const [formData, setFormData] = useState<LedgerRegistrationData>({
+    name: '',
+    phoneNumber: '',
+    initialAmount: '',
+    receptionist: '',
+    paymentMethod: '',
+  });
+
+  const [showReceptionistDropdown, setShowReceptionistDropdown] =
+    useState(false);
+  const [showPaymentMethodDropdown, setShowPaymentMethodDropdown] =
+    useState(false);
+
+  // 접수자 옵션
+  const receptionistOptions = [
+    '홍길동',
+    '김직원',
+    '이사장',
+    '박매니저',
+    '최대리',
+  ];
+
+  // 결제수단 옵션
+  const paymentMethodOptions = ['현금', '계좌이체'];
+
+  const handleConfirm = () => {
+    // 모든 필드가 입력되었는지 확인
+    if (
+      !formData.name.trim() ||
+      !formData.phoneNumber.trim() ||
+      !formData.initialAmount.trim() ||
+      !formData.receptionist ||
+      !formData.paymentMethod
+    ) {
+      return;
+    }
+
+    onConfirm(formData);
+    handleClose();
+  };
+
+  const handleClose = () => {
+    // 폼 데이터 초기화
+    setFormData({
+      name: '',
+      phoneNumber: '',
+      initialAmount: '',
+      receptionist: '',
+      paymentMethod: '',
+    });
+    setShowReceptionistDropdown(false);
+    setShowPaymentMethodDropdown(false);
+    onClose();
+  };
+
+  const formatAmount = (text: string) => {
+    // 숫자만 남기기
+    const cleaned = text.replace(/\D/g, '');
+
+    // 천 단위 콤마 추가
+    return cleaned.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+  };
```

Comment:
```suggestion
  const formatAmount = (amount: string) => {
    const formattedAmount = amount.replace(/\D/g, '').replace(/\B(?=(\d{3})+(?!\d))/g, ',');
    return formattedAmount;
  };
```
이렇게도 할 수도 있을 것 같은데, 사실 복잡한 정규식은 바로 알아보기 어려우니 
const ADD_COMMA_REGEX 같이 변수 선언해서 constant / regex.ts에 따로 관리하는 방법도 좋고,
아니면 더 쉽게 알아볼 수 있게 정규식 대신 다른 방식을 쓰는것도 좋음